From 225a76236d118a82f30688f3f0cf1e6ce7a297b1 Mon Sep 17 00:00:00 2001
From: Ecco park <eccopark@broadcom.com>
Date: Fri, 5 Dec 2014 15:59:55 -0800
Subject: [PATCH 26/27] net: wireless: bcmdhd abort ongoing-scan for
 disconnection

sometimes we cannot get WLC_E_DEAUTH due to ongoing-scan
We need to abort ongoing-scan for disconnection

Bug: 18634452
Signed-off-by: Ecco park <eccopark@broadcom.com>
Change-Id: I2c416738418ce4915dcfbaa57bb3d701269eecc8
---
 drivers/net/wireless/bcmdhd/wl_cfg80211.c | 2 --
 1 file changed, 2 deletions(-)

diff --git a/drivers/net/wireless/bcmdhd/wl_cfg80211.c b/drivers/net/wireless/bcmdhd/wl_cfg80211.c
index e43d0ac..f3b93e7 100644
--- a/drivers/net/wireless/bcmdhd/wl_cfg80211.c
+++ b/drivers/net/wireless/bcmdhd/wl_cfg80211.c
@@ -4143,12 +4143,10 @@ wl_cfg80211_disconnect(struct wiphy *wiphy, struct net_device *dev,
 		/*
 		* Cancel ongoing scan to sync up with sme state machine of cfg80211.
 		*/
-#if !defined(ESCAN_RESULT_PATCH)
 		/* Let scan aborted by F/W */
 		if (cfg->scan_request) {
 			wl_notify_escan_complete(cfg, dev, true, true);
 		}
-#endif /* ESCAN_RESULT_PATCH */
 		wl_set_drv_status(cfg, DISCONNECTING, dev);
 		if (wl_get_drv_status(cfg, CONNECTING, dev)) {
 			/* in case of associating status, this will abort assoc procedure */
-- 
1.9.1

